datafusion-examples/Cargo.toml

```diff
@@ -66,6 +66,7 @@ datafusion-expr = { workspace = true }
 datafusion-functions-window-common = { workspace = true }
 datafusion-optimizer = { workspace = true, default-features = true }
 datafusion-physical-expr = { workspace = true, default-features = true }
+datafusion-physical-optimizer = { workspace = true, default-features = true }
```

Comment:
This just follows the existing patterns so I think it is ok, but I think in general the examples should only be using the  DataFusion crate (to ensure all relevant structures are exposed)

I will make a follow on PR

Reply:
- https://github.com/apache/datafusion/pull/14305